docs/src/modules/java/pages/quickstart/cr-value-entity-kafka-java.adoc

```diff
@@ -0,0 +1,160 @@
+----
+
+. Add the service definition. The service definition is annotated with `akkaserverless.codegen` indicating we want to generate an Action for this service.
+. Add declarations for:
++
+* Listening to `customer.domain.CustomerState` events from the value entity `customer` by using the option `eventing.in`.
+* Publishing to the Kafka topic `customer_upserts` by using the option `eventing.out`.
++
+[source,proto,indent=0]
+.src/main/proto/customer/action/customer_action.proto
+----
+include::example$java-customer-registry-kafka-quickstart/src/main/proto/customer/action/customer_action.proto[tag=service]
+----
+
+. Run `mvn compile` from the project root directory to generate source classes in which you add business logic.
++
+----
+mvn compile
+----
+
+== Create an Action
+
+Actions are stateless functions that can be triggered in multiple ways. In this case the action is triggered by each event `customer.domain.CustomerState` received by the Value Entity `customer.domain.Customer`. 
+
+. If it's not open already, open `src/main/java/customer/action/CustomerStateSubscriptionAction.java` for editing.
+
+. Modify the `onUpsertState` method by adding the logic to handle the action. The complete method should include the following:
++
+[source, java]
+.src/main/java/customer/action/CustomerStateSubscriptionAction.java
+----
+include::example$java-customer-registry-kafka-quickstart/src/main/java/customer/action/CustomerStateSubscriptionAction.java[tag=upsert]
+----
++
+* The incoming message contains the request data from your client and this action just passes it along to the xref:java:actions-publishing-subscribing.adoc[Pub/Sub] mechanism of choice. In this example Kafka.
+
+== Package and deploy your service
+
+To compile, build the container image, and publish it to your container registry, follow these steps
+
+. From the root project directory, compile the source code using Maven:
++
+[source, command line]
+----
+mvn compile
+----
+
+. Use the `deploy` target to build the container image and publish it to your container registry. At the end of this command Maven will show you the container image URL you'll need in the next part of this quickstart.
++
+[source, command line]
+----
+mvn deploy
+----
+
+. Sign in to your Akka Serverless account at: {console}
+. If you do not have a project, click *Add Project* to create one, otherwise choose the project you want to deploy your service to.
+. On the project dashboard click the "+" next to *services* to start the deployment wizard
+. Choose a name for your service and click *Next*
+. Enter the container image URL from the above step and click *Next*
+. Click *Next* (no environment variables are needed for these samples)
+. Check both _Add a route to this service_ and _Enable CORS_ and click *Next*
+. Check your message broker is configured to use Kafka by following the https://developer.lightbend.com/docs/akka-serverless/projects/message-brokers.html[Configure message brokers] how-to.
```

Comment:
Configuring the message broker will be a prerequisite for the service to start, so having and configuring it should probably be in the "Before you begin" section

Reply:
done